samtranslator/model/api/api_generator.py

```diff
@@ -925,6 +926,11 @@ def _set_endpoint_configuration(self, rest_api, value):
         :param rest_api: RestApi resource
         :param string/dict value: Value to be set
         """
-
-        rest_api.EndpointConfiguration = {"Types": [value]}
-        rest_api.Parameters = {"endpointConfigurationTypes": value}
+        if isinstance(value, dict) and value.get("Type"):
+            rest_api.Parameters = {"endpointConfigurationTypes": value.get("Type")}
+            rest_api.EndpointConfiguration = {"Types": [value.get("Type")]}
+            if "VPCEndpointIds" in value.keys():
```

Comment:
I didn't see this earlier, but I wished it was `VpcEndpointIds` instead of `VPCEndpointIds`, just for consistency. :) I did not realise and kept trying with `VpcEndpointIds` and could not get it to work (there's no error, so it deployed successfully, without VPCe).

Reply:
I agree with @saurabhshri. Would  be better if the name can be made consistence as per cloudformation EndpointConfiguration properties.

This one confuses us, especially, because the documentation says the wrong thing.
https://docs.aws.amazon.com/serverless-application-model/latest/developerguide/sam-resource-api.html#sam-api-endpointconfiguration